### PR TITLE
Add weekly digest messsages for no additional notes and no pending v…

### LINKED ISSUE #2387: Add Explanatory Text When There is No Data for Sections of Supervisor E-mail
**What type(s) of user does this feature affect?**
 - supervisors
 
**Description**
Currently, when the supervisor receives their weekly e-mail it contains sections for Additional Notes and Pending Volunteers which can be empty. We would like for these sections to, if they do not contain other data, display the messages "There are no additional notes." and "There are no pending volunteers." respectively. 

**Screenshots of current behavior, if any**  
Supervisor Email

_QA Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 123456

### PR DESCRIPTION
**Pull request by colefortner**
### What github issue is this PR for, if any?
Resolves [#2387](https://github.com/rubyforgood/casa/issues/2387)

### What changed, and why?

Explanatory text is visible when there is no data to display in the Additional Notes and Pending Volunteers section of the weekly digest email sent to supervisors

### How is this tested? (please write tests!) 💖💪

```
      it "does not display 'There are no pending volunteers' message when there are pending volunteers" do
        expect(mail.body.encoded).to_not match("There are no pending volunteers.")
      end
    end

    it "displays no pending volunteers message when there are no pending volunteers" do
      expect(mail.body.encoded).to match("There are no pending volunteers.")
    end

    it "does not display 'There are no additional notes' message when there are additional notes" do
      create(:case_assignment, casa_case: casa_case, volunteer: volunteer, active: false, updated_at: Date.today + 8.days)
      expect(mail.body.encoded).to_not match("There are no additional notes.")
    end

    it "displays no additional notes messsage when there are no additional notes" do
      expect(mail.body.encoded).to match("There are no additional notes.")
    end
```

### Screenshots please :)

WeeklyDigestPREmail

### Feelings gif (optional)

First PR

### CODE REVIEW

File: spec/mailers/supervisor_mailer_spec.rb [hunk 1]
```diff
@@ -82,6 +82,24 @@
...
+        expect(mail.body.encoded).to_not match("There are no pending volunteers.")
+      end
+    end
+
+    it "displays no pending volunteers message when there are no pending volunteers" do
+      expect(mail.body.encoded).to match("There are no pending volunteers.")
+    end
+
+    it "does not display 'There are no additional notes' message when there are additional notes" do
+      # this_week = Date.today - 7.days..Date.today
```
Comment (compwron) [inline spec/mailers/supervisor_mailer_spec.rb:96]: ok to delete commented out line

**Comment by compwron** [comment]
@colefortner this is looking very good! I think we will be ready to merge very soon! 
TODO for you:
1. fix the merge conflict
1. add some screenshot/s to the PR + fill in PR text
1. mark PR as Ready For Review

**Comment by colefortner** [comment]
Ok, thanks!  I will work on that after my class tonight.

**Comment by compwron** [comment]
@colefortner  This is looking good! You have some failing tests to fix and then we will MERGE! :)
```
spec/views/supervisor_mailer/weekly_digest.html.erb_spec.rb
spec/mailers/supervisor_mailer_spec.rb
```

**Review by compwron — APPROVED** [review]

**Comment by compwron** [comment]
Tests are passing, let's MERGE!

**Pull request merged**